swarm/src/behaviour.rs

```diff
@@ -163,21 +171,31 @@ pub trait NetworkBehaviour: 'static {
     ) {
     }
 
-    /// Informs the behaviour about a closed connection to a peer.
-    ///
-    /// A call to this method is always paired with an earlier call to
-    /// [`NetworkBehaviour::inject_connection_established`] with the same peer ID, connection ID and endpoint.
-    fn inject_connection_closed(
-        &mut self,
-        _: &PeerId,
-        _: &ConnectionId,
-        _: &ConnectedPoint,
-        _: <Self::ConnectionHandler as IntoConnectionHandler>::Handler,
-        _remaining_established: usize,
-    ) {
-    }
+    // TODO: Have to remove this as it takes ownership of `Handler`.
```

Comment:
You could call `on_event` in the default implementation and construct the corresponding variant of `InEvent`.

Reply:
Genius @thomaseizinger! A lot better. Done with https://github.com/libp2p/rust-libp2p/pull/2867/commits/de5c46dbe198c756c27de467196324084dee86b8.